mod_app/admin/film_admin.py

```diff
@@ -96,6 +98,23 @@ def safe_comments(self, obj):
     safe_comments.allow_tags = True
     safe_comments.short_description = "Comments"
 
+    def preview_video(self, obj):
+        if obj.videos:
```

Comment:
You have a conflict between the videos property you defined on the Film model (which I think you're referring to here) and the videos related name which is created by the Video class (due to the related_name argument on the film field).

I think the problem is that the related manager is being tested here, which returns True even when there are no videos.

Reply:
I've changed it to if `obj.videos.first()` but if it would be better to do it another way that's simple then I'm happy to do that

Reply:
Why not rename the videos property to something like linked_videos? Otherwise the conflict is still there and it might trip you up in future.

Reply:
Actually, I don't think you even need to set a property since obj.videos.all() should be equivalent to what you get with the property.

Reply:
Same goes for the rest of the properties on Film pointing to link models: programmes, pub_mats, stills, postcards etc.